lib/uri/common.rb

```diff
@@ -340,16 +330,36 @@ def self.regexp(schemes = nil)
   #   and then to encoding +enc+.
   #
   # In either case, the returned string has forced encoding Encoding::US_ASCII.
-  #
   def self.encode_www_form_component(str, enc=nil)
     _encode_uri_component(/[^*\-.0-9A-Z_a-z]/, TBLENCWWWCOMP_, str, enc)
   end
 
-  # Decodes given +str+ of URL-encoded form data.
+  # Returns a string derived from the given \URL-encoded string +str+.
+  # Before decoding, +str+ is force-encoded to +enc+.
```

Comment:
Looking at the source code of `_decode_uri_component`, `str` has its encoding first changed to `Encoding::ASCII-8BIT` (using `String#b`), then the conversion is done, then is force-encoded to `enc`.

Reply:
Reworded.